presto-iceberg/src/test/java/com/facebook/presto/iceberg/hive/TestRenameTableOnFragileFileSystem.java

```diff
@@ -0,0 +1,606 @@
+
+            assertFalse(fileSystem.exists(new Path(newTableMetadataPath)));
+            assertFalse(fileSystem.exists(new Path(newTablePermissionDirPath)));
+            assertFalse(fileSystem.exists(new Path(newTablePermissionFilePath)));
+            assertTrue(fileSystem.exists(new Path(originTableMetadataPath)));
+            assertTrue(fileSystem.exists(new Path(originTablePermissionDirPath)));
+            assertTrue(fileSystem.exists(new Path(originTablePermissionFilePath)));
+
+            try {
+                icebergHiveMetadata.renameTable(connectorSession, icebergTableHandle, new SchemaTableName(newSchemaName, newTableName));
+            }
+            catch (Exception e) {
+                fail("Rename table should not fail: " + e.getMessage());
+            }
+
+            schemaTableNames = icebergHiveMetadata.listTables(connectorSession, Optional.of(newSchemaName));
+            assertTrue(schemaTableNames.contains(new SchemaTableName(newSchemaName, newTableName)));
+            schemaTableNames = icebergHiveMetadata.listTables(connectorSession, Optional.of(newSchemaName));
+            assertFalse(schemaTableNames.contains(new SchemaTableName(originSchemaName, originTableName)));
+
+            assertTrue(fileSystem.exists(new Path(newTableMetadataPath)));
+            assertTrue(fileSystem.exists(new Path(newTablePermissionDirPath)));
+            assertTrue(fileSystem.exists(new Path(newTablePermissionFilePath)));
+            assertFalse(fileSystem.exists(new Path(originTableMetadataPath)));
+            assertFalse(fileSystem.exists(new Path(originTablePermissionDirPath)));
+            assertFalse(fileSystem.exists(new Path(originTablePermissionFilePath)));
+        }
+        finally {
+            fileSystem.delete(new Path(originSchemaDirPath), true);
+            fileSystem.delete(new Path(newSchemaDirPath), true);
+        }
+    }
+
+    @Test
+    public void testRenameTableSucceedWithDeleteRedundantPermissionFileFails()
+            throws Exception
+    {
+        FileHiveMetastoreConfig config = createFileHiveMetastoreConfig();
+        TestingHdfsEnvironment hdfsEnvironment = getTestingHdfsEnvironment();
+        FileHiveMetastore metastore = new FileHiveMetastore(hdfsEnvironment, config);
+        IcebergHiveMetadata icebergHiveMetadata = (IcebergHiveMetadata) getIcebergHiveMetadata(metastore);
+        ExtendedFileSystem fileSystem = hdfsEnvironment.getFileSystem(connectorSession.getUser(), new Path(originSchemaMetadataPath), new Configuration());
+        try {
+            createFile(fileSystem, new Path(originSchemaMetadataPath), DATABASE_CODEC.toBytes(databaseMetadata));
+            createFile(fileSystem, new Path(newSchemaMetadataPath), DATABASE_CODEC.toBytes(databaseMetadata));
+            createFile(fileSystem, new Path(originTableMetadataPath), TABLE_CODEC.toBytes(tableMetadata));
+            createFile(fileSystem, new Path(originTablePermissionFilePath), new byte[128]);
+
```

Comment:
Let's extract this common code to a method use that instead in all the places where these file setup is done to reduce repetition